### PR TITLE
Participant readme

### PR DESCRIPTION
**Pull request by lashd**
This may or may not be ready but thought I would raise the request so that we can start to discuss the participant readme as required by #39.

@TheWonn has used these instructions and then go on to the exercises.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,58 @@
...
-## What's in the Bootcamp
-The course is being actively built out so be sure to pull regularly to get new exercises as they appear.
-### Infrastructure as code
-#### Ansible
-#### Testing
-## Getting Started
+This course is intended to teach good practices around Infrastructure-as-Code development. This course contains a set of materials which are designed to teach you how to write tests for and automate the release of the things you build.
+
+To undertake this course you will need a computer which you have full control over (i.e not your employer's desktop build) - It will need to be on a network with unrestricted access to the Internet (not behind locked-down corporate firewalls) as you will need to download and install a few components before you can begin.
+
```
Comment (jimconner) [inline README.md]: Is it worth adding a minimum spec paragraph here... pointing out that the course uses both virtualisation and containerisation and therefore has reasonably high memory requirements. Also require Intel VT or AMD-V virtualisation extensions turned on in the bios to support Virtualbox.

File: .templates/README.md.erb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+To remove the need for dependencies on an external IaaS (Infrastructure as a Service) solution in order to participate in this course, this course has been designed to use Docker containers to simulate real computers.
+
+The course is being actively built out so be sure to `git pull` regularly to get new exercises as they appear.
+
+## Participants
+### Environment Requirements
+For this material you will need the following:
+ - A Bash compatible environment
+ - git
```
Comment (jimconner) [inline .templates/README.md.erb]: - with a web browser

File: .templates/README.md.erb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ - Docker - If you don't have it already, follow the instructions on the [Docker website](https://docs.docker.com/install/#next-release) to install it on your system
+ - Editor of your choice
+
+### Setup
+1. create a .netrc file in your home directory with your credentials for github in it.
+  Your entry should look something like this:
+  ```
+  machine api.github.com
+      login username
```
Comment (jimconner) [inline .templates/README.md.erb]: Let's use the convention that required fields are wrapped in tie fighters <my_username>, <my_password> etc

File: .templates/README.md.erb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      password password
+  ```
+2. Fork this repo
```
Comment (jimconner) [inline .templates/README.md.erb]: Maybe a link to the github docs on forking would be an idea hear... People won't all necessarily be au-fait with git based workflows... https://help.github.com/articles/fork-a-repo/

File: bin/cic [hunk 5]
```diff
@@ -3,6 +3,8 @@
 CURRENT_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 # shellcheck source=bin/support/sh/functions/cic_docker.sh
 source "${CURRENT_PATH}/support/sh/functions/cic_docker.sh"
+# shellcheck source=bin/support/sh/functions/cic.sh
```
Comment (jimconner) [inline bin/cic]: Nice - I like that you've made the effort to tell shellcheck where to find the files we're sourcing in.

File: bin/support/sh/functions/cic.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+if [ "${TRACKS_PATH}" == "" ]
+then
+    TRACKS_PATH="${CURRENT_PATH}/../tracks"
+fi
+
```
Comment (jimconner) [inline bin/support/sh/functions/cic.sh]: We can use bash's default varaible features here and avoid an if statement by using:  `TRACKS_PATH=${TRACKS_PATH:-${CURRENT_PATH}/../tracks}`  

instead... and likewise for EXERCISES_PATH below too.

File: .templates/README.md.erb (hunk 2 above)
Comment (lashd) [reply .templates/README.md.erb]: Agreed

File: bin/support/sh/functions/cic.sh (hunk 6 above)
Comment (lashd) [reply bin/support/sh/functions/cic.sh]: Thanks, this is really useful!

File: README.md (hunk 1 above)
Comment (lashd) [reply README.md]: Worth mentioning if you think that it's likely to happen and the help given by VirtualBox and the likes is not going to be very helpful. Could you put together a paragraph?

**Pull request merged**